@here/olp-sdk-dataservice-read/test/unit/StreamLayerClient.test.ts

```diff
@@ -484,4 +504,76 @@ describe("StreamLayerClient", () => {
                 expect(err.message).to.be.equal(mockError);
             });
     });
+
+    it("Should seek return error if mode is parallel and subscriptionId is missed", async () => {
+        const mockOffsets = {
+            offsets: [
+                {
+                    partition: 1,
+                    offset: 1
+                },
+                {
+                    partition: 2,
+                    offset: 2
+                }
+            ]
+        };
+        const mockError =
+            "Error: for 'parallel' mode 'subscriptionId' is required.";
+
+        const request = new dataServiceRead.SeekRequest()
+            .withMode("parallel")
+            .withSeekOffsets(mockOffsets);
+        const seek = await streamLayerClient.seek(request).catch(err => {
+            assert.isDefined(err);
+            expect(err.message).to.be.equal(mockError);
+        });
+    });
+
+    it("Should seek return error if offsets are missed", async () => {
+        const mockError = "Error: offsets are required.";
+
+        const request = new dataServiceRead.SeekRequest();
+        const seek = await streamLayerClient.seek(request).catch(err => {
+            assert.isDefined(err);
+            expect(err.message).to.be.equal(mockError);
+        });
+    });
+
+    it("Should seek set offsets and return OK", async () => {
+        const headers = new Headers();
+        headers.append("X-Correlation-Id", "9141392.f96875c-9422-4df4-bdfj");
+        const mockResponse = ({
+            headers,
+            json: function() {
+                return this;
+            },
+            status: 200
+        } as unknown) as Response;
+        const mockOffsets = {
+            offsets: [
+                {
+                    partition: 1,
+                    offset: 1
+                },
+                {
+                    partition: 2,
+                    offset: 2
+                }
+            ]
+        };
+        seekOffsetsStub.callsFake(
+            (builder: any, params: any): Promise<Response> => {
+                return Promise.resolve(mockResponse);
+            }
+        );
+
+        const request = new dataServiceRead.SeekRequest().withSeekOffsets(
+            mockOffsets
+        );
+        const unsubscription = await streamLayerClient.seek(request);
+
+        assert.isDefined(unsubscription);
+        expect(unsubscription.status).to.be.equal(mockResponse.status);
```

Comment:
Could you please add a test for calling StreamApi.seekToOffset method with expected arguments and another one test for checking correct updating this.xCorrelationId after the response from the service? Thanks.

Reply:
Added